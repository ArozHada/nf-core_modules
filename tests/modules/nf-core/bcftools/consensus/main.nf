#!/usr/bin/env nextflow

nextflow.enable.dsl = 2
moduleDir = launchDir

include { BCFTOOLS_CONSENSUS } from "$moduleDir/modules/nf-core/bcftools/consensus/main.nf"

workflow test_bcftools_consensus {

    input = [
        [ id:'test' ], // meta map
        file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
        file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true),
        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
    ]

    BCFTOOLS_CONSENSUS ( input )
}
