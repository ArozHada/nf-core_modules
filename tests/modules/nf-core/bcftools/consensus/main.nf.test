nextflow_process {

    name "Test Process BCFTOOLS_CONSENSUS"
    script "modules/nf-core/bcftools/consensus/main.nf"
    process "BCFTOOLS_CONSENSUS"
    config "tests/modules/nf-core/bcftools/consensus/nextflow.config"
    tag "bcftools_consensus"

    test("Sarscov2 Illumina VCF Genome Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test' ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }
    }
}
