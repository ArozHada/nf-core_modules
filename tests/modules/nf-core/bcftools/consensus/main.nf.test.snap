{
    "Sarscov2 Illumina VCF Genome Fasta": {
        "content": [
            [
                "test.fa:md5,e57d3e4d113f989bc069c5fd61627091",
                "versions.yml:md5,12d82a78ae17363a1d697e785a08469f"
            ]
        ],
        "timestamp": "2023-04-08T15:22:25+0000"
    }
}